mlir/tools/rocmlir-tuning-driver/rocmlir-tuning-driver.cpp

```diff
@@ -288,13 +288,17 @@ static LogicalResult runTuningLoop(ModuleOp source) {
       return failure();
     }
 
-    // Extract binary anb benchmark
+    // Extract binary and benchmark
     std::string hipModule;
-    tuneCopy.walk([&hipModule](gpu::GPUModuleOp op) {
-      hipModule = op->getAttrOfType<StringAttr>("gpu.binary").getValue().str();
-      return WalkResult::interrupt();
+    tuneCopy.walk([&hipModule, &kernelFuncName](gpu::GPUModuleOp op) {
+      std::string moduleName = op.getName().str();
```

Comment:
I there there might be a way to avoid `.str()`-ing here with `op.getName()` and then, in the `if` path, using a `Twine`, but that's not entirely clear.
It's not a massive thing, but it'll save us some memory allocations.

Reply:
probably I can do 
if (moduleName == Twine(kernelFunc.getSymName(), "_module").str())

Reply:
Do you need the `.str()` there? I forget if there's a `StringRef == Twine`

Reply:
(if you do then never mind)

Reply:
... actually, I'd think to do this
(and this is with `moduleName` staying a `StringRef`)
```
if (!moduleName.endsWith("_module")) {
  llvm::errs() << "GPU module arising from rocmlir pipeline has unexpected name format: << moduleName << "\n";
  [abort the tuning]
}
if (moduleName.drop_back(8) == funcName) {
  ...
}
```